tango_simlib/examples/override_class.py

```diff
@@ -777,8 +799,52 @@ def action_slew(self, model, tango_dev=None, data_input=None):
         model.sim_quantities['desiredPointing'].set_val(
             [data_input[1], data_input[2]], model_time)
 
-    def pre_update(self, sim_model):
-        MODULE_LOGGER.info("***Updating from the override class***")
+    def post_update(self, sim_model, sim_time, dt):
+        MODULE_LOGGER.info("***Post-updating from the override class***")
+
+    def pre_update(self, sim_model, sim_time, dt):
+        MODULE_LOGGER.info("***Pre-updating from the override class***")
+
+        pointing_state_quant = sim_model.sim_quantities['pointingState']
+        current_pnt_state_enum_val = pointing_state_quant.last_val
+        current_pnt_state_str_val = (
+            pointing_state_quant.meta['enum_labels'][int(current_pnt_state_enum_val)])
+
+        if current_pnt_state_str_val == 'READY':
```

Comment:
Can't you just work with the enum directly? Perhaps add another dict to the quant meta called 'enum_vals', then you can just do:
```
point_state_enum =  pointing_state_quant.meta['enum_vals']
current_pnt_state_enum_val == point_state_enum['READY']
```

Perhaps not for this pull request, but it is something we should think about, since all the code dealing with enums look quite clunky.